opensrp-core/src/main/java/org/smartregister/util/StatsUtils.java

```diff
@@ -52,77 +53,84 @@ public StatsUtils() {
+            while (cursor.moveToNext()) {
+                populateEventSyncInfo(cursor);
+            }
+            cursor.close();
 
-        cursor = database.rawQuery(validatedEventsSql, new String[]{});
-        while (cursor.moveToNext()) {
-            populateValidatedEventsInfo(cursor);
-        }
-        cursor.close();
+            cursor = database.rawQuery(clientSyncSql, new String[]{});
+            while (cursor.moveToNext()) {
+                populateClientSyncInfo(cursor);
+            }
+            cursor.close();
 
-        cursor = database.rawQuery(validatedClientsSql, new String[]{});
-        while (cursor.moveToNext()) {
-            populateValidatedClientsInfo(cursor);
-        }
+            cursor = database.rawQuery(validatedEventsSql, new String[]{});
+            while (cursor.moveToNext()) {
+                populateValidatedEventsInfo(cursor);
+            }
+            cursor.close();
 
-        cursor = database.rawQuery(unsyncedVaccineEventsSQL, new String[]{});
-        while (cursor.moveToNext()) {
-            syncInfoMap.put(UNSYNCED_VACCINE_EVENTS, String.valueOf(cursor.getInt(0)));
-        }
+            cursor = database.rawQuery(validatedClientsSql, new String[]{});
+            while (cursor.moveToNext()) {
+                populateValidatedClientsInfo(cursor);
+            }
 
-        cursor = database.rawQuery(unsyncedWeightEventsSQL, new String[]{});
-        while (cursor.moveToNext()) {
-            syncInfoMap.put(UNSYNCED_WEIGHT_EVENTS, String.valueOf(cursor.getInt(0)));
-        }
-        if (CoreLibrary.getInstance().context().getAppProperties().isTrue("monitor.height")) { // Constant is defined in growth-monitoring module
-            cursor = database.rawQuery(unsyncedHeightEventsSQL, new String[]{});
+            cursor = database.rawQuery(unsyncedVaccineEventsSQL, new String[]{});
             while (cursor.moveToNext()) {
-                syncInfoMap.put(UNSYNCED_HEIGHT_EVENTS, String.valueOf(cursor.getInt(0)));
+                syncInfoMap.put(UNSYNCED_VACCINE_EVENTS, String.valueOf(cursor.getInt(0)));
             }
-        }
 
-        cursor.close();
+            cursor = database.rawQuery(unsyncedWeightEventsSQL, new String[]{});
+            while (cursor.moveToNext()) {
+                syncInfoMap.put(UNSYNCED_WEIGHT_EVENTS, String.valueOf(cursor.getInt(0)));
+            }
+            if (CoreLibrary.getInstance().context().getAppProperties().isTrue("monitor.height")) { // Constant is defined in growth-monitoring module
+                cursor = database.rawQuery(unsyncedHeightEventsSQL, new String[]{});
+                while (cursor.moveToNext()) {
+                    syncInfoMap.put(UNSYNCED_HEIGHT_EVENTS, String.valueOf(cursor.getInt(0)));
+                }
+            }
 
-        populateUserInfo();
-        populateBuildInfo();
-        populateDeviceInfo();
-        return syncInfoMap;
+            cursor.close();
```

Comment:
In case of any exceptions we can add the `cursor.close()` call in a `finally{` block to avoid memory leaks